source/particle/generator/uniform_radial.cc

```diff
@@ -0,0 +1,250 @@
+                const double theta_spacing = (P_max[2] - P_min[2]) / fmax(theta_particles-1,1);
+
+                for (unsigned int j = 0; j < theta_particles; ++j)
+                  {
+                    spherical_coordinates[2] = P_min[2] + j * theta_spacing;
+
+                    //Average value of sin(n) from 0 to 180 degrees is (2/pi)
+                    const unsigned int adjusted_phi_particles = std::max(static_cast<unsigned int> (phi_particles * std::sin(spherical_coordinates[2])),(unsigned int) 1);
+                    const double phi_spacing = (P_max[1] - P_min[1]) / fmax(adjusted_phi_particles-1,1);
+                    for (unsigned int k = 0; k < adjusted_phi_particles; ++k)
+                      {
+                        spherical_coordinates[1] = P_min[1] + k * phi_spacing;
+                        const Point<dim> newPoint = Utilities::cartesian_coordinates<dim>(spherical_coordinates) + P_center;
+                        particles.insert(this->generate_particle(newPoint,cur_id++));
+                      }
+                  }
+              }
+            else
+              ExcNotImplemented();
+          }
+
+        return particles;
+      }
+
+
+      template <int dim>
+      void
+      UniformRadial<dim>::declare_parameters (ParameterHandler &prm)
+      {
+        prm.enter_subsection("Postprocess");
+        {
+          prm.enter_subsection("Tracers");
+          {
+            prm.declare_entry ("Number of tracers", "1000",
+                               Patterns::Double (0),
+                               "Total number of tracers to create (not per processor or per element). "
+                               "The number is parsed as a floating point number (so that one can "
+                               "specify, for example, '1e4' particles) but it is interpreted as "
+                               "an integer, of course.");
+
+            prm.enter_subsection("Generator");
+            {
+              prm.enter_subsection("Uniform radial");
+              {
+                prm.declare_entry ("Minimal radius", "0",
+                                   Patterns::Double (),
+                                   "Minimal radial coordinate for the region of tracers. "
+                                   "Measured from the center position.");
+                prm.declare_entry ("Maximal radius", "1",
+                                   Patterns::Double (),
+                                   "Maximal radial coordinate for the region of tracers. "
+                                   "Measured from the center position.");
+                prm.declare_entry ("Minimal longitude", "0",
+                                   Patterns::Double (),
+                                   "Minimal longitude coordinate for the region of tracers. "
+                                   "Measured from the center position.");
```

Comment:
what is "center" in this context? same for the other parameters below.

You measure longitude and latitude in radians. Is this common? Wouldn't using degrees make more sense? (You'd simply convert in the one place where you read the parameter.)